compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -93,6 +101,27 @@ pub(super) fn search_for_section<'a>(
         .map_err(|e| format!("failed to read {} section in '{}': {}", section, path.display(), e))
 }
 
+pub(super) fn get_metadata_xcoff<'a>(path: &Path, data: &'a [u8]) -> Result<&'a [u8], String> {
+    let buf = search_for_section(path, data, ".info")?;
```

Comment:
How do we know that the first entry in the `.info` section is our metadata? Are we sure that rustc is the only source of data in `.info`? According to the [documentation](https://www.ibm.com/docs/en/aix/7.2?topic=formats-xcoff-object-file-format#XCOFF__s1b3i3e0jbau), we should be using a symbol to locate it instead:
> The n_name field is preserved by the binder. An application-defined unique name in this field can be used to filter access to only those comment section strings intended for the application.

Reply:
For some object files, multiple segments may contain sections with the same name. However as I know, with XCOFF there may be multiple sections with type `info`, but their names must be different. Therefore, the ".info" section can be specific for metadata.

Reply:
My understanding is that `.info` may contain multiple comments, and we need to use a symbol to locate the comment in `.info` that rustc added.

Reply:
I see.
We've added a symbol with value of 4, which specifies the offset in comment section,  to refer to the rust metadata.
```
    // Add a symbol referring to .info section.
    // Storage class will be handled in object crate.
    file.add_symbol(Symbol {
        name: "info".into(),
        value: 4,
        size: 0,
        kind: SymbolKind::Unknown,
        scope: SymbolScope::Dynamic,
        weak: false,
        section: SymbolSection::Section(section),
        flags: SymbolFlags::Xcoff { n_sclass : xcoff::C_INFO },
    });
 ```
 Do you mean that we should use the value in the symbol to locate the metadata instead of reading the first entry directly? In the meanwhile, we can change the name 'info' to 'rustc' to make the symbol be the identifier?

Reply:
Based on the documentation, yes.